lib/api/v3/work_packages/form/work_package_payload_representer.rb

```diff
@@ -108,6 +121,27 @@ def work_package_attribute_links_representer(represented)
           def description_renderer
             ::API::Utilities::Renderer::TextileRenderer.new(represented.description, represented)
           end
+
+          def parse_date_only(value, property_name)
+            return nil if value.nil?
+
+            begin
+              date_and_time = DateTime.iso8601(value)
+            rescue ArgumentError
+              raise API::Errors::PropertyFormatError.new(property_name, 'YYYY-MM-DD', value)
+            end
+
+            date_only = date_and_time.to_date
+
+            # we only want to accept "timeless" dates, e.g. "2015-01-31",
+            # but not "2015-01-31T01:02:03".
+            # However Date.iso8601 is too generous and would accept that
+            unless date_and_time == date_only
+              raise API::Errors::PropertyFormatError.new(property_name, 'YYYY-MM-DD', value)
+            end
+
+            date_only
+          end
```

Comment:
@NobodysNightmare I'd like to get the payload representer as small as possible. For that I propose moving the date parsing to some helper object that all representers (or whoever) can reuse. Otherwise I see the danger of us having to reimplement this time and time again in the various payload representers and then having to switch the format on the date string.

Because of said format, it should ideally also include the getter part. That would also take care of the repetition of
```<date attribute>.to_date.iso8601 unless <date attribute>.nil?```
in four places in this PR alone.

And if that is not enough for you to buy it, it will also cut down the required specs.